src/librustc_mir/monomorphize/collector.rs

```diff
@@ -511,6 +512,15 @@ impl<'a, 'tcx> MirNeighborCollector<'a, 'tcx> {
 }
 
 impl<'a, 'tcx> MirVisitor<'tcx> for MirNeighborCollector<'a, 'tcx> {
+    fn visit_body(&mut self, body: &Body<'tcx>) {
+        for const_ in &body.uneval_consts {
+            let location = mir::START_BLOCK.start_location();
```

Comment:
this may cause some diagnostics regressions, not sure.

Reply:
can you check all implementors of `visit_constant` and how they use `location`? We may be able to remove the argument (separate PR in that case) or we'll have to start storing the location together with the constant in the vector.

Reply:
I've added a "fake" location because if I'm not wrong it's not used.
See https://github.com/rust-lang/rust/pull/70820/files#diff-d583e4efe1a72516e274158e53223633L592

Reply:
in that case we can remove the argument from the visitor :)

Reply:
I meant that is not used in this particular implementation of the visitor. There are cases like https://github.com/rust-lang/rust/blob/6b158c6f75f75e19221952ea24c43725b856a34e/src/librustc_mir/borrow_check/type_check/mod.rs#L297 that uses the location.

Reply:
oh. I completely failed to realize that this is not the `walk_body` method. I think we should/must run this for all visitors by default. So you should move this impl to the `Visitor` trait impl

Reply:
If the constant references a static, that static shouldn't be codegened, only evaluated.

Reply:
Going to resolve this conversation too because this code is not in the PR anymore.

Reply:
Imo this code addition should be in `super_body` in the generic visitor and not in the collector.

Reply:
:+1: